lesson_16/homework_16_1.py

```diff
@@ -0,0 +1,177 @@
+"""ДЗ 16.1. Ромбовидне наслідування та Геометрична задача.
+
+Завдання 1.
+
+Створіть клас Employee, який має атрибути name та salary.
+Далі створіть два класи, Manager та Developer, які успадковуються від Employee.
+Клас Manager повинен мати додатковий атрибут department, а клас Developer -
+атрибут programming_language.
+Тепер створіть клас TeamLead, який успадковується як від Manager,
+так і від Developer. Цей клас представляє керівника з команди розробників.
+Клас TeamLead повинен мати всі атрибути як Manager (ім'я, зарплата, відділ),
+а також атрибут team_size, який вказує на кількість розробників у команді,
+якою керує керівник.
+Напишіть тест, який перевіряє наявність атрибутів з Manager та Developer у
+класі TeamLead
+"""
+
+import math
+from abc import ABC, abstractmethod
+
+
+# Клас Employee
+class Employee:
+    """Клас Employee з атрибутами name та salary."""
+
+    def __init__(self, name, salary):
+        """Ініціалізує працівника з ім'ям та зарплатою."""
+        self.name = name
+        self.salary = salary
+
+
+class Manager(Employee):
+    """Клас Manager, успадковується від Employee."""
+
+    def __init__(self, name, salary, department):
+        """Ініціалізує менеджера з ім'ям, зарплатою та відділом."""
+        Employee.__init__(self, name, salary)
+        self.department = department
+
+
+class Developer(Employee):
+    """Клас Developer, успадковується від Employee."""
+
+    def __init__(self, name, salary, programming_language):
+        """Ініціалізує розробника з ім'ям, зарплатою та мовою програмування."""
+        Employee.__init__(self, name, salary)
+        self.programming_language = programming_language
+
+
+class TeamLead(Manager, Developer):
+    """Клас TeamLead, успадковується від Manager та Developer."""
+
+    def __init__(
+            self, name, salary, department, programming_language, team_size,
+    ):
+        """Ініціалізує керівника команди."""
+        Manager.__init__(self, name, salary, department)
```

Comment:
Please use single super() call to use full power of inheritance . Here is example how to do this 
https://github.com/antohaUa/AQA_241007_Hillel/blob/main/Lesson_17_import/00_prev_inheritance.py